### ISSUE TITLE
ClassCastException when ResultQuery.fetchArray() returns no rows

### ISSUE THREAD

**Issue by luStopai** [jOOQ/jOOQ#11521]
Hi

I think I've found a bug.

When making a manual request that returns RecordN a ClassCastException occurs if the request returns no rows.

I had something like 

```java
Record3<Integer, String, Boolean>[] records = dsl.select(field1, field2, field3).from(...).where(...).fetchArray();
```

But when the request returns no rows, I think it tries to return and empty array of Record0 which generates a ClassCastException when trying to cast it to an empty array of Record3 elements.

### Versions

- jOOQ: 3.14.7 (through spring-boot-starter-jooq 2.4.3)
- Java: 11
- Database (include vendor): postgres
- OS: linux
- JDBC Driver (include name if inofficial driver): 42.2.18

**Comment by lukaseder** [comment]
Related issue: https://github.com/jOOQ/jOOQ/issues/11208

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce this. Will fix right away for 3.15.0 and 3.14.8

**Comment by lukaseder** [comment]
This is a leftover from this change in 3.14.0: https://github.com/jOOQ/jOOQ/issues/4695. See `SelectQueryImpl::getRecordType0`:

```java
// TODO: [#4695] Calculate the correct Record[B] type
else
    return (Class<? extends R>) RecordImplN.class;
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.15.0 and 3.14.8 (#11526). Thanks again for your report!
